crates/fuel-core/src/query/balance/asset_query.rs

```diff
@@ -99,12 +104,25 @@ impl<'a> AssetsQuery<'a> {
                     } else {
                         return Err(anyhow::anyhow!("The coin is not UTXO").into());
                     };
-                    // TODO: Fetch coin in a separate thread
-                    let coin = self.database.coin(id)?;
-
-                    Ok(CoinType::Coin(coin))
+                    Ok(id)
                 })
+            });
+
+        futures::stream::StreamExt::chunks(stream, database.batch_size)
```

Comment:
could this all be cleaned up with the new `yield_each` to avoid the need to map chunks and flatten results?

Reply:
No, here we actually send a batch request to the database like: "Please fetch the all coins for these list of UtxoIds".

Later we can optimize it with multi get + caching

Reply:
Oic, since it might be multiget later?

Reply:
Yep